### PR TITLE
[BottomSheet] Example unit tests for updatePreferredSheetHeight - WIP

### PR DESCRIPTION
**Pull request by romoore**
A couple of basic unit tests to verify the logic flows inside `updatePreferredSheetHeight`. They could definitely benefit from some refactoring within MDCBottomSheetPresentationController, but given the current code structure these seem to be the simplest way to exercise the new code independently.

Marked "WIP" so nobody tries to merge this as it's just an example to illustrate my thoughts on #5023

### CODE REVIEW

File: components/BottomSheet/tests/unit/MDCBottomSheetPresentationControllerTests.m [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+- (void)testPreferredSheetHeightWhenPreferredContentSizeNonZero {
+  // Given
+  FakeSheetView *sheetView = [[FakeSheetView alloc] init];
+  UIViewController *presentingViewController = [[UIViewController alloc] init];
+  MDCFakePresentedViewController *presentedViewController = [[MDCFakePresentedViewController alloc] init];
+  MDCFakeBottomSheetPresentationController *presentationController =
+      [[MDCFakeBottomSheetPresentationController alloc]
+          initWithPresentedViewController:presentedViewController
+          presentingViewController:presentingViewController];
+  presentationController.test_PresentedViewController = presentedViewController;
```
Comment (romoore) [inline components/BottomSheet/tests/unit/MDCBottomSheetPresentationControllerTests.m]: This extra assignment/property may not even be needed given that the initializer probably just assigns the property.

Comment (romoore) [reply components/BottomSheet/tests/unit/MDCBottomSheetPresentationControllerTests.m]: Yeah, it can.

**Comment by codeman7** [comment]
I don't see why not to merge. It adds the functionality and test that we need.

File: components/BottomSheet/tests/unit/BottomSheetPresentationTest.swift [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+  func testSetPreferredSheetHeight() {
+    // Given
+    let preferredSheetHeight: CGFloat = 125.0
+
+    // When
+    presentationController.preferredSheetHeight = preferredSheetHeight
+    presentationController.presentedViewController.view.setNeedsLayout()
+    presentationController.presentedViewController.view.layoutIfNeeded()
+    print(presentationController.frameOfPresentedViewInContainerView)
```
Comment (codeman7) [inline components/BottomSheet/tests/unit/BottomSheetPresentationTest.swift:40]: Please remove before merge.

**Review by codeman7 — APPROVED** [review]

**Comment by romoore** [comment]
Made obsolete by #5115

**Pull request closed without merging**